web/html/src/manager/minion/ansible/ansible-control-node.tsx

```diff
@@ -0,0 +1,227 @@
+import * as React from "react";
+import SpaRenderer from "core/spa/spa-renderer";
+import { Messages, Utils } from "components/messages";
+import { TextField } from "components/fields";
+import { Panel } from "components/panels/Panel";
+import { AsyncButton, Button } from "components/buttons";
+import Network from "utils/network";
+import NewAnsiblePath from "./new-ansible-path";
+import EditAnsiblePath from "./edit-ansible-path";
+
+type AnsiblePath = {
+  id?: number;
+  minionServerId?: number;
+  type?: string;
+  path?: string;
+}
+
+type PropsType = {
+  minionServerId: number;
+};
+
+type StateType = {
+  minionServerId: number;
+  playbooksPaths: AnsiblePath[];
+  inventoriesPaths: AnsiblePath[];
+  newPlaybookPath: string;
+  newInventoryPath: string;
+  editPlaybookPath?: AnsiblePath;
+  editInventoryPath?: AnsiblePath;
+  errors: string[];
+};
+
+class AnsibleControlNode extends React.Component<PropsType, StateType> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      minionServerId: props.minionServerId,
+      playbooksPaths: [],
+      inventoriesPaths: [],
+      newPlaybookPath: "",
+      newInventoryPath: "",
+      editPlaybookPath: undefined,
+      editInventoryPath: undefined,
+      errors: [],
+    };
+
+    Network.get("/rhn/manager/api/systems/details/ansible/paths/" + props.minionServerId)
+    .promise.then(data => {
+      this.setState({ playbooksPaths: data.filter(p => p.type === "playbook"), inventoriesPaths: data.filter(p => p.type === "inventory") });
+    });
+  }
+
+  newPath(type: string, newPath: string) {
+    if (type === "playbook") {
+      this.setState({ newPlaybookPath: newPath });
+    }
+    else {
+      this.setState({ newInventoryPath: newPath });
+    }
+  }
+
+  deletePath(path: AnsiblePath) {
+    Network.post(
+      "/rhn/manager/api/systems/details/ansible/paths/delete",
+      JSON.stringify(path.id),
```

Comment:
Small nitpick, if this is just a number, I would recommend `.toString()` instead since then it's clear that we're not dealing with any JSON here.